code/builders/builder-webpack5/src/preview/iframe-webpack.config.ts

```diff
@@ -243,6 +243,13 @@ export default async (
     },
     resolve: {
       extensions: ['.mjs', '.js', '.jsx', '.ts', '.tsx', '.json', '.cjs'],
+      // This mapping is taken from TypeScript's file extension substitution logic
+      // for more info see https://www.typescriptlang.org/docs/handbook/modules/reference.html#file-extension-substitution
+      extensionAlias: {
+        '.js': ['.ts', '.tsx', '.d.ts', '.js', '.jsx'],
+        '.mjs': ['.mts', '.d.mts', '.mjs'],
+        '.cjs': ['.cts', '.d.cts', '.cjs'],
```

Comment:
Perhaps the declaration file extensions are not needed?

```suggestion
        '.js': ['.ts', '.tsx', '.js', '.jsx'],
        '.mjs': ['.mts',  '.mjs'],
        '.cjs': ['.cts',  '.cjs'],
```